src/proto/stream.rs

```diff
@@ -201,11 +200,9 @@ where
 
                     self.queue_frame(frame);
                 }
-                // SAFETY: match statement at the start of the method ensures that this is not the
-                // case
-                StreamState::ClosedByPeer | StreamState::CloseAcknowledged => unsafe {
-                    unreachable_unchecked()
-                },
+                StreamState::ClosedByPeer | StreamState::CloseAcknowledged => {
+                    debug_assert!(false, "unexpected StreamState");
```

Comment:
This branch basically gets removed and falls through the below `_ => {}` branch

Reply:
Seems like it, this is a very trivial optimization.